opm/material/common/UniformTabulated2DFunction.hpp

```diff
@@ -185,22 +186,31 @@ class UniformTabulated2DFunction
     /*!
      * \brief Evaluate the function at a given (x,y) position.
      *
-     * If this method is called for a value outside of the tabulated
-     * range, a \c Opm::NumericalIssue exception is thrown.
+     * \param x x-position
+     * \param y y-position
+     * \param extrapolate Whether to extrapolate for untabulated values. If false then
+     * an exception might be thrown.
      */
     template <class Evaluation>
-    Evaluation eval(const Evaluation& x, const Evaluation& y) const
+    Evaluation eval(const Evaluation& x, const Evaluation& y, bool extrapolate) const
     {
-#ifndef NDEBUG
         if (!applies(x,y))
         {
-            throw NumericalIssue("Attempt to get tabulated value for ("
-                                   +std::to_string(double(scalarValue(x)))+", "+std::to_string(double(scalarValue(y)))
-                                   +") on a table of extend "
-                                   +std::to_string(xMin())+" to "+std::to_string(xMax())+" times "
-                                   +std::to_string(yMin())+" to "+std::to_string(yMax()));
+            std::string msg = "Attempt to get tabulated value for ("
+                +std::to_string(double(scalarValue(x)))+", "+std::to_string(double(scalarValue(y)))
+                +") on a table of extend "
```

Comment:
Possible follow-up here: Is this supposed to say "extent" instead?

Reply:
not from me but fixes anyway